### PR TITLE
MEN-4818 Monitoring add-on: initial docs

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:title
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

**Comment by merlin-northern** [comment]
@estenberg I could really use an early review here. if the direction is correct I will continue along these lines.

File: 02.Overview/01.Introduction/docs.md [hunk 1]
```diff
@@ -11,7 +11,8 @@ architecture allowing central management of deployments to all devices.
 Mender also supports powerful extensions in a form of
 [add-ons](../../09.Add-ons/00.Overview/docs.md) which 
 enable features like Remote Terminal, Port Forward, File Transfer,
-and Device Configuration.
+and Device Configuration, together with the ability to monitor
+and alert with the Monitoring add-on.
```
Comment (estenberg) [inline 02.Overview/01.Introduction/docs.md:15]: Monitoring -> Monitor

File: 09.Add-ons/00.Overview/docs.md [hunk 2]
```diff
@@ -8,8 +8,9 @@ taxonomy:
 ## Mender Connect
 
 The add-ons are independent applications, but they run under a common executable:
-`mender-connect`. Every add-on, with [configure](../02.Configure/docs.md) being one notable
-exception, requires [Mender Connect](../../09.Downloads/docs.md#mender-connect) to function.
+`mender-connect`. Every add-on, with [Configure](../02.Configure/docs.md)
```
Comment (estenberg) [inline 09.Add-ons/00.Overview/docs.md:11]: Would reword this, there are too many exceptions. E.g. "The Troubleshoot add-on package requires `mender-connect`"

Also Monitoring -> Monitor

File: 09.Add-ons/00.Overview/docs.md [hunk 3]
```diff
@@ -22,6 +23,7 @@ The following table shows a brief summary of the add-ons.
 | [File Transfer](../03.File-Transfer/docs.md) | Upload and download files to and from a device with both UI and CLI | Troubleshoot |
 | [Port Forward](../40.Port-Forward/docs.md) | Forward any local port to a port on a device without opening ports on the device | Troubleshoot |
 | [Configure](../02.Configure/docs.md) | Apply configuration to your devices through a uniform interface | Configure |
+| [Monitoring](../50.Monitoring/docs.md) | Monitor your devices and get alerts | Monitoring |
```
Comment (estenberg) [inline 09.Add-ons/00.Overview/docs.md:26]: Monitor

File: 09.Add-ons/50.Monitoring/docs.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
+---
+title: Monitoring
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:2]: Monitor

File: 09.Add-ons/50.Monitoring/docs.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+taxonomy:
+    category: docs
+    label: user guide
+---
+
+!!!!! The Mender Monitoring add-on package is required.
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:8]: Monitor

File: 09.Add-ons/50.Monitoring/docs.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+!!!!! See [the Mender features page](https://mender.io/plans/features?target=_blank)
+!!!!! for an overview of all Mender plans and features.
+
+The Mender Monitoring add-on gives the power to
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:12]: Monitor

File: 09.Add-ons/50.Monitoring/docs.md [hunk 7]
```diff
@@ -0,0 +1,123 @@
...
+    category: docs
+    label: user guide
+---
+
+!!!!! The Mender Monitoring add-on package is required.
+!!!!! See [the Mender features page](https://mender.io/plans/features?target=_blank)
+!!!!! for an overview of all Mender plans and features.
+
+The Mender Monitoring add-on gives the power to 
+You do not need `mender-connect`
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:13]: Would skip this.

File: 09.Add-ons/50.Monitoring/docs.md [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+command:
+```bash
+systemctl start mender-monitor
+```
+The service does normal logging which you can see with the `journalctl -u mender-monitor`
+command, but also stores a copy of logs in `/var/log/mender-monitord.log` file.
+Keep in mind to log-rotate that file as needed.
+
+### Monitoring subsystems
+The `mender-monitor` supports _monitoring subsystems_ which perform
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:29]: Remove The

File: 09.Add-ons/50.Monitoring/docs.md [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+    label: user guide
+---
+
+!!!!! The Mender Monitoring add-on package is required.
+!!!!! See [the Mender features page](https://mender.io/plans/features?target=_blank)
+!!!!! for an overview of all Mender plans and features.
+
+The Mender Monitoring add-on gives the power to 
+You do not need `mender-connect` 
+It consists of two parts: the API to the backend that allows
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:14]: Move this text to the next section, or remove it.

Add 1-2 sentences about what's the point of this Monitor add-on. Why should I use it? What's the use cases / user value?

File: 09.Add-ons/50.Monitoring/docs.md [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+you to integrate with existing solutions, and a `mender-monitor`
+systemd service being an out-of-the-box monitoring daemon
+
+## Running and configuring
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:18]: This is good information, but maybe a bit too advanced to cover this early? Could be covered in a separate "Architecture" sub-page or further down in the document I think.

Maybe rewrite this one to two examples:
* How do I ensure the systemd service "mender-client" is running?
* How do I check for a pattern "input device disconnected" in the kernel journal log?

**Review by estenberg — CHANGES_REQUESTED** [review]
Nothing wrong with the direction, just try to keep it as simple as possible for the initial tutorial.
Also remember we need similar subsections like Remote Terminal, (Installation, Configuration) -- the latter you may perhaps cover the more advanced topics (subsystems, etc.).

File: 09.Add-ons/50.Monitoring/docs.md (hunk 7 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:13]: wow -- sorry some unfinished thougts:)

File: 09.Add-ons/50.Monitoring/docs.md (hunk 10 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:18]: moved to Architecture section and added examples section. with mender-connect check (if mender-client is not running we can't alert we do not have the JWT) and `Started Session [^ ]+ of user root` pattern in /var/log/messages (I could not find the exact example with the device in my logs)

File: 09.Add-ons/50.Monitoring/docs.md (hunk 9 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:14]: yes, that was the idea behind the unfinished sentence. let me fix it.

File: 09.Add-ons/50.Monitoring/docs.md (hunk 8 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:29]: fixed.

File: 09.Add-ons/50.Monitoring/docs.md (hunk 6 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:12]: fixed

File: 09.Add-ons/50.Monitoring/docs.md (hunk 4 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:2]: fixed

File: 09.Add-ons/00.Overview/docs.md (hunk 3 above)
Comment (merlin-northern) [reply 09.Add-ons/00.Overview/docs.md:26]: fixed

File: 09.Add-ons/00.Overview/docs.md (hunk 2 above)
Comment (merlin-northern) [reply 09.Add-ons/00.Overview/docs.md:11]: reworded, also added a column  to the table below.

File: 02.Overview/01.Introduction/docs.md (hunk 1 above)
Comment (merlin-northern) [reply 02.Overview/01.Introduction/docs.md:15]: fixed.

File: 09.Add-ons/50.Monitoring/docs.md [hunk 11]
```diff
@@ -0,0 +1,173 @@
...
+    category: docs
+    label: user guide
+---
+
+!!!!! The Mender Monitor add-on package is required.
+!!!!! See [the Mender features page](https://mender.io/plans/features?target=_blank)
+!!!!! for an overview of all Mender plans and features.
+
+The Mender Monitor add-on gives you an insight to what happens
+within your fleet of devices. It allows you to configure and receive
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:13]: allows you to configure alerts to be sent when certain

File: 09.Add-ons/50.Monitoring/docs.md [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+taxonomy:
+    category: docs
+    label: user guide
+---
+
+!!!!! The Mender Monitor add-on package is required.
+!!!!! See [the Mender features page](https://mender.io/plans/features?target=_blank)
+!!!!! for an overview of all Mender plans and features.
+
+The Mender Monitor add-on gives you an insight to what happens
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:12]: gives you instant visibility into issues with your device fleet

File: 09.Add-ons/50.Monitoring/docs.md [hunk 13]
```diff
@@ -0,0 +1,173 @@
...
+within your fleet of devices. It allows you to configure and receive
+alerts when certain parts of your infrastructure are malfunctioning
+or need your attention. We designed it as a lightweight and distributed
+solution for the world of the IoT.
+
+## Architecture
+The Monitor add-on consists of two parts: the API to the backend that allows
+you to integrate with existing solutions, and a `mender-monitor`
+systemd service being an out-of-the-box monitoring daemon.
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:21]: daemon, that runs on the devices.

File: 09.Add-ons/50.Monitoring/docs.md [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+The former is a set of bash functions, which you can call 
+from wrappers or use at your own discretion, while the latter
+is a simple service that can easily be configured to send alerts
+to the Mender Server when certain event occur.
+
+### Running and configuring
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:27]: How about installation? Is it covered separately (would make sense)? if so, we need a link here to how to install it first.

File: 09.Add-ons/50.Monitoring/docs.md [hunk 15]
```diff
@@ -0,0 +1,173 @@
...
+
+!!!!! The Mender Monitor add-on package is required.
+!!!!! See [the Mender features page](https://mender.io/plans/features?target=_blank)
+!!!!! for an overview of all Mender plans and features.
+
+The Mender Monitor add-on gives you an insight to what happens
+within your fleet of devices. It allows you to configure and receive
+alerts when certain parts of your infrastructure are malfunctioning
+or need your attention. We designed it as a lightweight and distributed
+solution for the world of the IoT.
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:16]: To save bandwidth and support edge processing, Monitor is a fully distributed solution, meaning that no data is sent from the device to the server *unless* an alert is triggered. Compared to classic approaches of sending all the data about all the nodes to the server all the time, this leads to massive bandwidth savings and allows instant local processing.

File: 09.Add-ons/50.Monitoring/docs.md [hunk 16]
```diff
@@ -0,0 +1,173 @@
...
+| Variable name   | Value       | Description |
+| --------------- | ----------- | ----------- |
+|ALERT_URL| string | URL to send the alerts to |
+|MONITOR_D_DIRECOTRY | string | a path to the `monitor.d` directory |
+|CONFIG_LOG_LEVEL| FATAL TRACE ERROR DEBUG WARN INFO | log level |
+|DAEMON_LOG_FILE| string | path for the daemon log file |
+|FLAPPING_INTERVAL| positive integer | length of the time interval for which flapping rate is calculated |
+|FLAPPING_COUNT_THRESHOLD| positive integer | number of service state changes per FLAPPING_INTERVAL above which service is considered to be flapping |
+
+## Alerting
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:164]: Move this as first section, before Architecture?

File: 09.Add-ons/50.Monitoring/docs.md [hunk 17]
```diff
@@ -0,0 +1,173 @@
...
+Each time a service is not running, or a log file contains a given pattern,
+all users that have access to a device receive an email notification.
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:166]: have access to the given device on the Mender server

File: 09.Add-ons/50.Monitoring/docs.md [hunk 18]
```diff
@@ -0,0 +1,173 @@
...
+```
+
+where `SERVICE_NAME` is arbitrary name used in logging and for identification,
+`LOG_PATTERN` is the regular expression we are trying to match to each
+line of the logs, and `LOG_FILE` is the path to the log file.
+
+The log subsystem saves the number of the last line of logs that it parsed,
+and starts tailing the file skipping the lines that it saw.
+
+### Daemon config defaults
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:148]: Nice and should be covered, but maybe better in a separate doc subsection?

**Review by estenberg — CHANGES_REQUESTED** [review]
Minor adjustments / nicpicks for readability. I think you've covered the meat here.

File: 09.Add-ons/50.Monitoring/docs.md (hunk 14 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:27]: I think we have to wait with that until we have a clear solution on how we distribute it. (the yocto external flow task)

File: 09.Add-ons/50.Monitoring/docs.md (hunk 11 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:13]: fixed

File: 09.Add-ons/50.Monitoring/docs.md (hunk 12 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:12]: fixed

File: 09.Add-ons/50.Monitoring/docs.md (hunk 13 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:21]: fixed

File: 09.Add-ons/50.Monitoring/docs.md (hunk 15 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:16]: added

File: 09.Add-ons/50.Monitoring/docs.md (hunk 16 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:164]: moved

File: 09.Add-ons/50.Monitoring/docs.md (hunk 17 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:166]: right.

File: 09.Add-ons/50.Monitoring/docs.md (hunk 18 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:148]: right! moved.

File: 09.Add-ons/50.Monitoring/30.Daemon-config/docs.md [hunk 19]
```diff
@@ -0,0 +1,22 @@
+---
+title: Monitoring configuration defaults
+taxonomy:
+    category: docs
+---
+
+
+### Daemon config defaults
+
+The daemon that the `mender-monitor` service runs has certain
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/30.Daemon-config/docs.md:10]: The `mender-monitor` daemon has default configurations set in `config/config.sh`.

Or did I misunderstand, is there a difference between the service and the daemon?

File: 09.Add-ons/50.Monitoring/30.Daemon-config/docs.md [hunk 20]
```diff
@@ -0,0 +1,22 @@
+---
+title: Monitoring configuration defaults
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/30.Daemon-config/docs.md:2]: Configuration

Also call the directory "30.Configuration" I think.

Even if there are other components to configure except the daemon, I would still keep them in this same document.

File: 09.Add-ons/50.Monitoring/docs.md [hunk 21]
```diff
@@ -0,0 +1,161 @@
...
+
+!!!!! The Mender Monitor add-on package is required.
+!!!!! See [the Mender features page](https://mender.io/plans/features?target=_blank)
+!!!!! for an overview of all Mender plans and features.
+
+The Mender Monitor add-on gives you instant visibility into issues
+with your device fleet. It allows you to configure alerts to be sent when certain
+parts of your infrastructure are malfunctioning
+or need your attention.
+To save bandwidth and support edge processing, Monitor is a fully
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:16]: Would add newline, so it's a separate paragraph.

File: 09.Add-ons/50.Monitoring/docs.md [hunk 22]
```diff
@@ -0,0 +1,161 @@
...
+distributed solution, meaning that we send no data from the device
+to the server unless triggered by an alert. Compared to classic approaches
+of sending all the data about all the nodes to the server all the time,
+this leads to massive bandwidth savings and allows instant local processing.
+
+## Alerting
+Each time a service is not running, or a log file contains a given pattern,
+all users that have access to the given device on the Mender server receive an email notification.
+You can mute the email notifications in the settings.
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:25]: in the Mender server settings.

File: 09.Add-ons/50.Monitoring/docs.md [hunk 23]
```diff
@@ -0,0 +1,161 @@
...
+
+### Running and configuring
+To start monitoring daemon it is enough to run the following
+command:
+```bash
+systemctl start mender-monitor
+```
+The service does normal logging which you can see with the `journalctl -u mender-monitor`
+command, but also stores a copy of logs in `/var/log/mender-monitord.log` file.
+Keep in mind to log-rotate that file as needed.
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:50]: Can we do this? It sounds a bit risky to have an unbounded log file. Or is this normally how log files behave (e.g. how do we do it with the Mender client?).

File: 09.Add-ons/50.Monitoring/docs.md [hunk 24]
```diff
@@ -0,0 +1,161 @@
...
+
+### Examples
+Before the detailed description of the add-on configuration and structure
+in details, we present two quick-start examples.
+
+#### Is mender-connect running?
+Assume you want to monitor the state of the `mender-connect` systemd service,
+and you want to receive alerts if the service is not running,
+as well as an _OK_ when it is back online. In order to get this
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:59]: "back online" -> "back up"

File: 09.Add-ons/50.Monitoring/docs.md [hunk 25]
```diff
@@ -0,0 +1,161 @@
...
+
+```bash
+ ln -s /etc/mender-monitor/monitor.d/available/service_menderconnect.sh /etc/mender-monitor/monitor.d/enabled/service_menderconnect.sh
+```
+
+The daemon will automatically pick the files and start checking the log file.
+
+
+#### Has user root started a session?
+Let's get the alerts when a `Started Session c244 of user root`-like log
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:77]: To alert on root sessions to the device, we check for the pattern `Started Session c244 of user root` in `/var/log/syslog`.

File: 09.Add-ons/50.Monitoring/docs.md [hunk 26]
```diff
@@ -0,0 +1,161 @@
...
+line appears in the `/var/log/syslog` file. Naturally we want to be aware
+only of new occurrences and the string between 'Session' and 'of'
+is something we do not know a priori.
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:80]: What is c244, a regex pattern? If so, add a short note about this.

E.g. is something we do not know a priori, so we use the PCRE pattern `c244` to match INSERT-HERE.

**Review by estenberg — CHANGES_REQUESTED** [review]
A few more nitpicks. :)

File: 09.Add-ons/50.Monitoring/30.Daemon-config/docs.md (hunk 19 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/30.Daemon-config/docs.md:10]: technicaly there is. fixed.

File: 09.Add-ons/50.Monitoring/30.Daemon-config/docs.md (hunk 20 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/30.Daemon-config/docs.md]: fixed

File: 09.Add-ons/50.Monitoring/docs.md (hunk 21 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:17]: added

File: 09.Add-ons/50.Monitoring/docs.md (hunk 22 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md]: fixed

File: 09.Add-ons/50.Monitoring/docs.md (hunk 23 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md]: my thoughts exactly, I think we should actually by default not to write to extra log-files. I will remove it and I created this task: https://tracker.mender.io/browse/MEN-4912

File: 09.Add-ons/50.Monitoring/docs.md (hunk 24 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md]: fixed

File: 09.Add-ons/50.Monitoring/docs.md (hunk 25 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md]: rephrased

File: 09.Add-ons/50.Monitoring/docs.md (hunk 26 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md]: no, it is a live log line from my rpi :) ok I will change it to a different log line.

File: 09.Add-ons/50.Monitoring/30.Configuration/docs.md [hunk 27]
```diff
@@ -0,0 +1,22 @@
+---
+title: Configuration
+taxonomy:
+    category: docs
+---
+
+
+### Daemon config defaults
+
+The `mender-monitor` daemon runs has certain
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/30.Configuration/docs.md:10]: remove runs

File: 09.Add-ons/50.Monitoring/docs.md [hunk 28]
```diff
@@ -0,0 +1,161 @@
...
+
+```bash
+ ln -s /etc/mender-monitor/monitor.d/available/service_menderconnect.sh /etc/mender-monitor/monitor.d/enabled/service_menderconnect.sh
+```
+
+The daemon will automatically pick the files and start checking the log file.
+
+
+#### Has a user started a session?
+To alert on new user sessions to the device, we can check for the pattern
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:77]: "new root user sessions"

File: 09.Add-ons/50.Monitoring/docs.md [hunk 29]
```diff
@@ -0,0 +1,161 @@
...
+`Started User Manager for UID 0` in the `/var/log/syslog` file.
+Naturally we want to be aware only of new occurrences and the number
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:79]: Remove this sentence.

File: 09.Add-ons/50.Monitoring/docs.md [hunk 30]
```diff
@@ -0,0 +1,161 @@
...
+after 'UID' is something we do not know a priori, we want the alert for every UID.
+We will need to create the following file:
+
+```bash
+ echo -e 'SERVICE_NAME="auth"\nLOG_PATTERN="Started User Manager for UID [0-9]"\nLOG_FILE="/var/log/syslog"' > /etc/mender-monitor/monitor.d/available/log_syslogrootsession.sh
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:84]: "...for UID 0" (dont make it a pattern)

File: 09.Add-ons/50.Monitoring/docs.md [hunk 31]
```diff
@@ -0,0 +1,161 @@
...
+and enable it by creating the following symbolic link:
+
+```bash
+ ln -s /etc/mender-monitor/monitor.d/available/service_menderconnect.sh /etc/mender-monitor/monitor.d/enabled/service_menderconnect.sh
+```
+
+The daemon will automatically pick the files and start checking the log file.
+
+
+#### Has a user started a session?
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:76]: Has root user started a session?

File: 09.Add-ons/50.Monitoring/docs.md [hunk 32]
```diff
@@ -0,0 +1,161 @@
...
+```bash
+ echo -e 'SERVICE_NAME="auth"\nLOG_PATTERN="Started User Manager for UID [0-9]"\nLOG_FILE="/var/log/syslog"' > /etc/mender-monitor/monitor.d/available/log_syslogrootsession.sh
+```
+
+and enable the check:
+
+```bash
+ ln -s /etc/mender-monitor/monitor.d/available/log_syslogrootsession.sh /etc/mender-monitor/monitor.d/enabled/log_syslogrootsession.sh
+```
+
```
Comment (estenberg) [inline 09.Add-ons/50.Monitoring/docs.md:92]: !!! You can also do PCRE pattern matching for the UID to catch users other than root.

**Review by estenberg — CHANGES_REQUESTED** [review]
If my suggestion is sane, I would change the example like this.
May be more relevant to catch root logins, in most cases it's probably OK to have other users log in.

File: 09.Add-ons/50.Monitoring/docs.md (hunk 32 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md:92]: added some beauty at the end of this sentence. fixed.

File: 09.Add-ons/50.Monitoring/docs.md (hunk 31 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md]: ok

File: 09.Add-ons/50.Monitoring/docs.md (hunk 30 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md]: ok but then we loose the whole glory of the regexes.

File: 09.Add-ons/50.Monitoring/docs.md (hunk 29 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md]: removed

File: 09.Add-ons/50.Monitoring/docs.md (hunk 28 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/docs.md]: changed

File: 09.Add-ons/50.Monitoring/30.Configuration/docs.md (hunk 27 above)
Comment (merlin-northern) [reply 09.Add-ons/50.Monitoring/30.Configuration/docs.md]: removed

**Review by estenberg — APPROVED** [review]
